preprocessing/__init__.py

```diff
@@ -115,4 +116,63 @@ def has_cycle(g: Graph, vertex: Vertex, predecessor: Vertex, visited):
         if v in visited and v is not predecessor:
             return True
         elif v not in visited:
-            return has_cycle(g, v, vertex, visited)
\ No newline at end of file
+            return has_cycle(g, v, vertex, visited)
+
+
+def get_modular_decomposition_sizes(md: ModularDecomposition):
+    return map(len, md)
+
+
+def check_modular_decomposition(md_g: ModularDecomposition, md_h: ModularDecomposition) -> bool:
+    """
+    Check if modular decompositions of two graphs indicate anisomorphism.
+
+    :param ModularDecomposition md_g: One modular decomposition.
+    :param ModularDecomposition md_h: Another modular decomposition.
+    :return: `False` if the graphs cannot be isomorphic; `True` otherwise.
+    """
+
+    return \
+        len(md_g) == len(md_h) \
+        and compare(get_modular_decomposition_sizes(md_g), get_modular_decomposition_sizes(md_h))
+
+
+def modular_decomposition_factor(md: ModularDecomposition) -> int:
```

Comment:
Was dit een gok of komt dit uit een paper?

Reply:
Ik vroeg me af of het niet misschien max. de factorial is. En of, als je 2 deze modules hebt (dus met 2 graven 2*2), of je dan niet meer krijgt.

Reply:
Ik snap je laatste zin niet helemaal.

Per module met grootte `m = |M|` heb je `m!` combinaties. Voor alle modules samen heb je `m_0! * m_1! * ... * m_n!` combinaties, met `n` het aantal modules. Ik denk niet dat dat het maximum is, maar dat dat gewoon het aantal combinaties is. Alle vertices in een module zijn onderling uitwisselbaar, want hun buren zijn dezelfde, dus je elke combinatie is een geldig isomorfisme.